src/hooks/use-search/use-search.test.tsx

```diff
@@ -0,0 +1,173 @@
+      const url = new URL(window.location.href);
+      window.history.replaceState({}, '', url);
+    });
+
+    it('sets state from the URL', () => {
+      const url = new URL(`${window.location.href}search?q=kafka&page=2`);
+      window.history.replaceState({}, '', url);
+
+      const { result } = renderHook(() =>
+        useSearch({
+          addsearchApiKey: 'key',
+          enableParamsSync: true,
+        }),
+      );
+
+      expect(result.current.state.query).toBe('kafka');
+      expect(result.current.state.page).toBe(2);
+    });
+  });
+
+  describe('search', () => {
+    it('calls search', () => {
+      const { result } = renderHook(() =>
+        useSearch({
+          addsearchApiKey: 'key',
+        }),
+      );
+
+      const searchMock = jest.fn();
+      jest.spyOn(result.current.state.client, 'search').mockImplementation(searchMock);
+
+      act(() => {
+        result.current.actions.search({ query: 'kafka', page: 2 });
+      });
+
+      expect(result.current.state.query).toBe('kafka');
+      expect(result.current.state.page).toBe(2);
+      expect(searchMock).toHaveBeenCalled();
+    });
+
+    it('resets state if query is deleted', () => {
+      const { result } = renderHook(() =>
+        useSearch({
+          addsearchApiKey: 'key',
+        }),
+      );
+
+      const searchMock = jest.fn();
+      jest.spyOn(result.current.state.client, 'search').mockImplementation(searchMock);
+
+      act(() => {
+        result.current.actions.search({ query: 'kafka', page: 2 });
+      });
+
+      expect(result.current.state.query).toBe('kafka');
+      expect(result.current.state.page).toBe(2);
+      expect(searchMock).toHaveBeenCalled();
+      searchMock.mockClear();
+
+      act(() => {
+        result.current.actions.search({ query: '' });
+      });
+
+      expect(result.current.state.query).toBe('');
+      expect(result.current.state.page).toBe(1);
+      expect(searchMock).not.toHaveBeenCalled();
+    });
+
+    it('sets loading state', () => {
+      const { result } = renderHook(() =>
+        useSearch({
+          addsearchApiKey: 'key',
+        }),
+      );
+
+      const searchMock = jest.fn();
+      jest.spyOn(result.current.state.client, 'search').mockImplementation(searchMock);
+
+      expect(result.current.state.loading).toBe(false);
+
+      act(() => {
+        result.current.actions.search({ query: 'kafka' });
+      });
+
+      expect(result.current.state.loading).toBe(true);
+    });
+
+    it('sets error if api returns error', () => {
+      const { result } = renderHook(() =>
+        useSearch({
+          addsearchApiKey: 'key',
+        }),
+      );
+
+      const searchMock = jest.fn((_, cb) => cb({ error: { message: 'Error' } }));
+      jest.spyOn(result.current.state.client, 'search').mockImplementation(searchMock);
+
+      act(() => {
+        result.current.actions.search({ query: 'kafka' });
+      });
+
+      expect(result.current.state.loading).toBe(false);
```

Comment:
Can this block of 4 expects be refactored into a helper method where the 4 expected values are passed into it.